core/src/main/scala/nl/vroste/zio/kinesis/client/Producer.scala

```diff
@@ -104,16 +132,86 @@ final case class ProducerSettings(
   failedDelay: Duration = 100.millis,
   metricsInterval: Duration = 30.seconds,
   updateShardInterval: Duration = 30.seconds,
-  aggregate: Boolean = false,
-  allowedErrorRate: Double = 0.05,
-  shardPredictionParallelism: Int = 8
+  shardPrediction: Producer.ShardPrediction = Producer.ShardPrediction.Enabled(),
+  aggregation: Producer.Aggregation = Producer.Aggregation.Disabled,
+  batchDuration: Option[Duration] = None
 ) {
-  require(allowedErrorRate > 0 && allowedErrorRate <= 1.0, "allowedErrorRate must be between 0 and 1 (inclusive)")
+  aggregation match {
+    case Producer.Aggregation.ByPredictedShard(_) =>
+      require(shardPrediction.isEnabled, "Aggregation requires shard prediction to be enabled")
+    case Producer.Aggregation.Disabled            => ()
+  }
 }
 
 object Producer {
+
+  sealed trait Throttling extends Product with Serializable
+
+  object Throttling {
+    case object Disabled extends Throttling
```

Comment:
I'm personally a fan of inlining the missing case into the sum type (and it should save an allocation), but we can also make allow of these options and get rid of disabled.

Reply:
`Throttling` is nice this way, let's keep it.